content/ADR-201-lambdas-wearables-API-analysis.md

```diff
@@ -0,0 +1,79 @@
+
+### API Functionality
+First, a review of the functionalities of the API will be covered. The API, based on the request filters, gets the wearables from the Content Server, applies a small transform and returns the result to the client.
+
+#### Request filters
+The API allows three query parameters:
+
+- `wearableIds`: A list of wearable URNs that the client requests. Just ask the Content Server for them and apply the transform. However, in some cases, when a large number of wearables are requested, the resulting URL string can exceed the maximum length allowed.
+- `collectionIds`: A list of collection URNs that the client requests. When a `collectionId` is "off-chain" (base-avatars only), it requests the wearables directly to the Content Server. If a `collectionId` is "on-chain" (collections v1 and v2), it runs queries on The Graph using the wearable URNs from that collection, and then it fetches them from the Content Server. No third-party collections are included. Note: for the base-avatars collection it has a fixed list in a file listing all the urns.
+- `textSearch`: A string value that is used to query The Graph. On-chain wearables have a metadata field called "searchText" that can have anything that is used for filtering. In some cases it has useful information but in other cases it has no meaningful info.
+
+#### Transform: Wearables from the Content Server to the Client
+
+1. First the wearable data from the Content Server is retrieved with this [schema](https://github.com/decentraland/schemas/blob/main/src/platform/entity.ts#L28-L37).
+2. Then only the `metadata` of type [Wearable](https://github.com/decentraland/schemas/blob/main/src/platform/item/wearable/wearable.ts#L18-L26) is returned to the client, but a small transform is applied: the contents are extended with full url.
+
+![Entity](/resources/ADR-201/entity-transform.png)
+
+### Explorer use cases of the API
+The API has many funcionalites, but the Explorer doesn't use all of them. After reviewing the `unity-renderer` repository, two use cases were recognized (check with Explorer Team): 
+1. `wearableIds`: The Explorer requests multiple wearables by URN. From the user profiles it reads the wearables that the user is wearing and request them. Note that the user can wear wearables from different collections.
+2. `collectionIds='urn:decentraland:off-chain:base-avatars'`: It uses the API functionality of requesting items from a collection but **always** for the base-avatar one. It should be because it is always needed for configuring the backpack where the base avatars are always available.
+
+
+## Solution Space Exploration
+
+### Solution: Explorer directly hits the Content Server
+
+Are these functionalities (strings concatenation mostly) currently worthy enough to have a handrail between the Explorer and Content Server through Lambdas?
```

Comment:
Can you confirm that the only transformation that is doing this handrail is to concatenate `peerContentUrl` before some fields (`thumbnail`, `image`, `contents`, etc.)?

Reply:
I have reasons to believe that in the past this Lambdas did more things as the Content Server didn't exposed the endpoint that i mentioned POST /entities/active. But now, with that new endpoint, it doesn't make much sense.